packages/rollup-contracts/contracts/L2ToL1MessageReceiver.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+/* Internal Imports */
+import {DataTypes as dt} from "./DataTypes.sol";
+
+contract L2ToL1MessageReceiver {
+    event L2ToL1MessageEnqueued(
+        address ovmSender,
+        bytes callData,
+        uint nonce
+    );
+
+    struct EnqueuedL2ToL1Message {
+        dt.L2ToL1Message message;
+        uint blockEnqueued;
+    }
+    
+    address public trustedSequencer;
+    uint public messageDelay;
+    uint messageNonce = 0;
+    mapping (uint => EnqueuedL2ToL1Message) public messages;
+
+    constructor(address _trustedSequencer, uint _messageDelay) public {
+        trustedSequencer = _trustedSequencer;
+        messageDelay = _messageDelay;
+    }
+
+    function enqueueL2ToL1Message(dt.L2ToL1Message memory _message) public {
+        require(msg.sender == trustedSequencer, "For now, only our trusted sequencer can enqueue messages to be verified on L1");
+        uint blockNum = block.number;
+        messages[messageNonce] = EnqueuedL2ToL1Message({
+            message: _message,
+            blockEnqueued: blockNum
+        });
+        messageNonce += 1;
+        emit L2ToL1MessageEnqueued(
```

Comment:
Will the event have a blocknum accessible by default? Also, will we want the nonce in this?

Reply:
good catch, that was supposed to be the nonce!